collection/list/variables.go

```diff
@@ -0,0 +1,13 @@
+package list
+
+type Collection struct {
+	FirstItem *Element
+	LastItem  *Element
+	len       int
+}
+
+type Element struct {
+	val      int
+	NextItem *Element
+	PrevItem *Element
```

Comment:
Type and its behavior are better to have in the same file by convention. So, type Element should be described in the collectionItem.go.